src/distro/distro.py

```diff
@@ -1078,6 +1082,47 @@ def uname_attr(self, attribute: str) -> str:
         """
         return self._uname_info.get(attribute, "")
 
+    def __resolve_chroot_symlink(self, link_location: str) -> str:
+        # only resolve symlink if root_dir is set and link is under it
+        if (
+            self.root_dir is None
+            or os.path.commonprefix([self.root_dir, link_location]) != self.root_dir
+        ):
+            return link_location
+
+        seen_paths = []
+        while True:
+            try:
+                resolved = os.readlink(link_location)
+            except OSError:  # includes case "not a symlink"
+                return link_location
+
+            if os.path.isabs(resolved):
+                # i.e. absolute path (regarding to the chroot), that we need to
+                # "move" back inside the chroot
+                resolved = os.path.splitdrive(resolved)[1].lstrip(os.sep)
+                if os.altsep is not None:
+                    resolved.lstrip(os.altsep)
+                resolved = os.path.join(self.root_dir, resolved)
+            else:
+                # i.e. relative path that we make absolute
+                resolved = os.path.join(os.path.dirname(link_location), resolved)
+
+            # remove redundant up-level references
+            resolved = os.path.normpath(resolved)
```

Comment:
My draft at https://github.com/python-distro/distro/pull/328#pullrequestreview-887955351 was avoiding `normpath` due to this sentence in the API docs:

>  This string manipulation may change the meaning of a path that **contains symbolic links**.

Emphasis mine, from https://docs.python.org/3/library/os.path.html#os.path.normpath .

Maybe this line can just be dropped?

Reply:
You were right, I ignored the documentation warning and that was a mistake. I added a case to triggering this.